### ISSUE TITLE
Improve Badge caching

### ISSUE THREAD

**Issue by rvalentini** [cljdoc/cljdoc#573]
The following behavior was reported when using Cljdoc badges on Github:
When refreshing the Github page shortly after a successful build of a newer release version, the link of the Cljdoc badge already resolves to the newer release version, whereas the Cljdoc badge SVG is still showing the old version. This is probably due to our caching configuration for badges.

* We are currently setting `max-age` to 30 min for badges: https://github.com/cljdoc/cljdoc/blob/master/src/cljdoc/server/pedestal.clj#L332
* To make things more complicated, Github is also doing its own caching: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-anonymized-urls#an-image-that-changed-recently-is-not-updating
* There were some previous issues with badgen.net that should be considered when changing the badge caching: https://github.com/cljdoc/cljdoc/issues/422

Maybe an option could be to introduce conditional requests by setting the `Last-Modified` or `ETag` response header for badges?

**Comment by corasaurus-hex** [comment]
It looks like we could use the [not-modified](http://pedestal.io/api/pedestal.service/io.pedestal.http.ring-middlewares.html#var-not-modified) [ring middleware](https://ring-clojure.github.io/ring/ring.middleware.not-modified.html) to do etags and lowering the `max-age` is pretty simple. I wonder how many more requests we'd be talking, though, and how the server would do under the extra load?

\<aside>
Coming at this from another angle: If we could throw something like Varnish in front of cljdoc we could control our own caching and evicting and serve a ton more reqs/s without breaking a sweat. I'm not sure that's something people would be up for, though.
\</aside>

I do like the idea of generating our own svg badges and I think we could probably do it fairly easily? https://github.com/lmarkus/openBadge

**Comment by lread** [comment]
Yeah, I thought delivering badges ourselves could be pretty straightforward too @corasaurus-hex: https://github.com/cljdoc/cljdoc/issues/422#issuecomment-808793684.

**Comment by corasaurus-hex** [comment]
Oooooh! great minds and all that 😉

**Comment by lread** [comment]
Related: I am seeing a sporadic 403 in prod when talking to badgen.net.

Sometimes it seems easier to reach for a 3rd party lib or service, but simpler to just implement the thing ourselves. After doing so, sometimes it can be surprising how easy it was to implement. 

I might explore that soon.
